### ISSUE TITLE
UC driver is super slow when launching

### ISSUE THREAD

**Issue by 4mekki4** [ultrafunkamsterdam/undetected-chromedriver#614]
Hello,

I have been using UC and selenium for the last weeks, and I noticed that UC is too slow when launching the driver (It takes 30 seconds or more) while the normal selenium launches in less than 10seconds. Is this normal?

**Comment by sebdelsol** [comment]
yes it downloads and patches the driver each time you instantiate one. 
That's the price to pay to have an undetected Chrome driver.

**Comment by 4mekki4** [comment]
> yes it downloads and patches the driver each time you instantiate `Chrome()`. That's the price to pay to have an undetected Chrome driver.

Is there a way to make a copy of the instantiated driver? or make the patches downloads happen one time per day or week.?

**Comment by sebdelsol** [comment]
You can use the `driver_executable_path` parameter. Please [check the docstring here](https://github.com/ultrafunkamsterdam/undetected-chromedriver/blob/b2e804e977c9cf3dd9c845474752cac142a72508/undetected_chromedriver/__init__.py#L142).

The driver won't be downloaded since you'll provide its path, and it'll only be patched once. 
But there are some caveats :
* Your driver will keep its signature and you might be detected because of that.
* It won't download the last driver version, so use a Chrome with the same version (use the `browser_executable_path` parameter for that).

**Comment by Honses111** [comment]
> You can use `driver_executable_path` argument for `uc.Chrome()` to provide a Chromedriver path. Please [check the docstring here](https://github.com/ultrafunkamsterdam/undetected-chromedriver/blob/b2e804e977c9cf3dd9c845474752cac142a72508/undetected_chromedriver/__init__.py#L142).
> 
> 
> 
> In this case, the driver won't be downloaded since you’ll provide it and it'll be patched only once. It’ll be as fast as Selenium… and it’ll skip undetected-chromedriver important features.
> 
> 
> 
> There are indeed some **caveats** :
> 
> * Your Chromedriver will keep its signature, please be aware you might be detected because of that.
> 
> * You won't download the latest Chromedriver version, so you’ll be stuck with its matching Chrome browser version. Use `browser_executable_path` argument to provide a non automatically updated matching Chrome browser executable.
> 
> * Do not instantiate multithreaded drivers : It'll throw random `IOError` because your driver still has to be concurrently opened to check it’s actually patched.

I admit that AutoDownloading and patching can avoid the same fingerprints formation by the same webdriver. However, I found that the recent downloaded and used webdriver won't be deleted automatically , they accumulate in the Folder of AppData\\Roaming\\undetected chromedriver in Windows. Fist of all, I don't notice this until this file become bulky enough and catch my attention.

**Comment by Honses111** [comment]
> The temp chromedriver file is actually deleted when the patched driver is freed. [Please check here](https://github.com/ultrafunkamsterdam/undetected-chromedriver/blob/b2e804e977c9cf3dd9c845474752cac142a72508/undetected_chromedriver/patcher.py#L250). 
> 
> Are you sure you exit your script properly ?

Thank you for your reply. Actually, I use multiprocess to repeatedly open and close several browser simultaneously in loops, may be because of this, most temp driver are locked and cannot deleted

**Comment by sebdelsol** [comment]
If you force `driver_executable_path` they won't be deleted but you might have random file error when the driver are concurrently opened to check they are patched.

If not those driver should actually be deleted except if they either fail or are killed or are terminated... 

Please run your script with the logger set to debug `logging.basicConfig(level=logging.DEBUG)` and check the logs.
